### PR TITLE
Fix documentation deployment

### PR DESCRIPTION
**Pull request by tobiasdiez**
### :books: Description

The update https://github.com/sagemath/sage/pull/35184 broke the docs upload. Hopefully this is fixed with this PR.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by dimpase** [comment]
OK, let's wait for CI to run its course

**Review by kwankyu — APPROVED** [review]
Okay, then.

**Review by dimpase — APPROVED** [review]

**Comment by tornaria** [comment]
Maybe it'd make sense to have an exception allowing workflow fixes like this one to be pushed to `develop`?

There are right now two critical PRs that are necessary for checks to pass: this one and #35418. The second one can be merged to a PR so checks are fixed, but this one cannot, as explained in https://github.com/sagemath/sage/pull/35356#issuecomment-1486425958.

It's quite unfortunate when checks are not reliable, since it leads us to learn to ignore failures. Right now the last PR with checks passing is #35352, so that's a lot of PRs without any chance at passing checks.

Some ideas for the future:
 - maybe releases could go throught a PR before being pushed to `devel` so there is confirmation that all checks pass.
 - PRs changing workflows need a special approval/testing since its changes are not tested by the normal CI.
 - an exception could be made for a small PR to be pushed to `devel` without waiting for a release if it is necessary to get checks to pass.

I would argue that both this one and #35418 should be pushed to `devel` right now. This one is clear as it only touches workflows. The other one is debatable, but the alternative is every PR merging #35418 and if one doesn't know or forgets checks will fail.

@vbraun

**Pull request merged**
